### PR TITLE
Added IWethGateway.sol

### PR DESCRIPTION
**Pull request by umair-mirza**
The Aave v2 documentation was missing the interface for Weth-Gateway. 
I've included the IWETHGateway.sol to the correct directory structure so that it can be included in the documentation.
The interface has been taken from this path: https://github.com/aave/protocol-v2/blob/1.0/contracts/misc/interfaces/IWETHGateway.sol

### CODE REVIEW

File: the-core-protocol/weth-gateway/iwethgateway.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+interface IWETHGateway {
+  function depositETH(
+    address pool,
+    address onBehalfOf,
+    uint16 referralCode
+  ) external payable;
+
+  function withdrawETH(
+    address pool,
```
Comment (je55g) [inline the-core-protocol/weth-gateway/iwethgateway.md:19]: "pool" --> "lendingPool"

File: the-core-protocol/weth-gateway/iwethgateway.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    uint256 amount,
+    address onBehalfOf
+  ) external;
+
+  function repayETH(
+    address pool,
```
Comment (je55g) [inline the-core-protocol/weth-gateway/iwethgateway.md:25]: "pool" --> "lendingPool"

File: the-core-protocol/weth-gateway/iwethgateway.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  ) external payable;
+
+  function borrowETH(
+    address pool,
+    uint256 amount,
+    uint256 interesRateMode,
+    uint16 referralCode
+  ) external;
+
+  function withdrawETHWithPermit(
```
Comment (je55g) [inline the-core-protocol/weth-gateway/iwethgateway.md:38]: Not a V2 function

File: the-core-protocol/weth-gateway/iwethgateway.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# IWETHGateway
+
+Also available on [Github](https://github.com/aave/protocol-v2/blob/1.0/contracts/misc/interfaces/IWETHGateway.sol).
```
Comment (je55g) [inline the-core-protocol/weth-gateway/iwethgateway.md:3]: https://github.com/aave/protocol-v2/blob/master/contracts/misc/interfaces/IWETHGateway.sol

**Review by je55g — COMMENTED** [review]
Changes needed as V3 "pool" and function have been used for V2

File: the-core-protocol/weth-gateway/iwethgateway.md (hunk 2 above)
Comment (umair-mirza) [reply the-core-protocol/weth-gateway/iwethgateway.md]: resolved.

File: the-core-protocol/weth-gateway/iwethgateway.md (hunk 3 above)
Comment (umair-mirza) [reply the-core-protocol/weth-gateway/iwethgateway.md]: resolved.

File: the-core-protocol/weth-gateway/iwethgateway.md (hunk 4 above)
Comment (umair-mirza) [reply the-core-protocol/weth-gateway/iwethgateway.md]: resolved.

**Comment by umair-mirza** [comment]
required fixes have been done.

**Review by je55g — APPROVED** [review]

**Pull request merged**